Earthfile

```diff
@@ -0,0 +1,112 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+FROM alpine:3.13
+WORKDIR /arrow
+
+# Declaring these makes Earthly pick them up from .env.
+ARG ARCH
+ARG ARCH_ALIAS
+ARG ARCH_SHORT_ALIAS
+ARG ULIMIT_CORE
+ARG REPO
+ARG CUDA
+ARG DEBIAN
+ARG UBUNTU
+ARG FEDORA
+ARG PYTHON
+ARG LLVM
+ARG CLANG_TOOLS
+ARG RUST
+ARG GO
+ARG NODE
+ARG MAVEN
+ARG JDK
+ARG NUMPY
+ARG PANDAS
+ARG DASK
+ARG TURBODBC
+ARG KARTOTHEK
+ARG HDFS
+ARG SPARK
+ARG DOTNET
+ARG R
+ARG ARROW_R_DEV
+ARG R_ORG
+ARG R_IMAGE
+ARG R_TAG
+ARG DEVTOOLSET_VERSION
+
+# Used for the manylinux and windows wheels
+ARG VCPKG=fced4bef1606260f110d74de1ae1975c2b9ac549
+
+conda-cpp:
+    FROM ./ci+conda-cpp-deps
+    # Alternative: use .dockerfile.
+    #FROM DOCKERFILE +dockerfile-context/ --dockerfile=conda-cpp
```

Comment:
This could be used to use the existing `.dockerfile` files (instead of the copy of those instructions within Earthly).

Reply:
Does it mandatory to use a context image for it or could we use the dockerfile directly from the filesystem?

Reply:
Changing an unrelated dockerfile would trigger a rebuild in conda-cpp?

Reply:
You can also use a Dockerfile directly from the system. However currently Earthly only supports Dockerfiles named `Dockerfile` and in the root of the build context. That's why it is repackaged like this.

In any case, we can add this support to Earthly. It has been requested before and it's not difficult to add.

> Changing an unrelated dockerfile would trigger a rebuild in conda-cpp?

It shouldn't - the command `COPY ./ci/docker+dockerfiles/"$dockerfile".dockerfile ./Dockerfile` should filter to the right dockerfile only.